File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        res = 0
        sum = neededTime[0]
        maxNum = sum
        for i in range(1, len(colors)):
            if colors[i] == colors[i - 1]:
                sum += neededTime[i]
                maxNum = max(maxNum, neededTime[i])
            else:
                res += sum - maxNum
                sum = neededTime[i]
                maxNum = sum
        res += sum - maxNum
        return res

File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int res = 0;
        int sum = neededTime[0];
        int maxNum = neededTime[0];
        for(int i = 1; i < colors.length(); i++)
        {   
            if(colors[i] == colors[i - 1])
            {
                sum += neededTime[i];
                maxNum = max(maxNum, neededTime[i]);
            }
            else
            {
                res += sum - maxNum;
                sum = neededTime[i];
                maxNum = neededTime[i];
            }
        }
        res += sum - maxNum;
        return res;
    }
};